adapters/tpmn/params_test.go

```diff
@@ -0,0 +1,45 @@
+package tpmn
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// TestValidParams makes sure that the tpmn schema accepts all imp.ext fields which we intend to support.
+func TestValidParams(t *testing.T) {
+	validator, err := openrtb_ext.NewBidderParamsValidator("../../static/bidder-params")
+	if err != nil {
+		t.Fatalf("Failed to fetch the json-schemas. %v", err)
+	}
+
+	for _, validParam := range validParams {
+		if err := validator.Validate(openrtb_ext.BidderTpmn, json.RawMessage(validParam)); err != nil {
+			t.Errorf("Schema rejected TPMN params: %s", validParam)
+		}
+	}
+}
+
+// TestInvalidParams makes sure that the tpmn schema rejects all the imp.ext fields we don't support.
+func TestInvalidParams(t *testing.T) {
+	validator, err := openrtb_ext.NewBidderParamsValidator("../../static/bidder-params")
+	if err != nil {
+		t.Fatalf("Failed to fetch the json-schemas. %v", err)
+	}
+
+	for _, invalidParam := range invalidParams {
+		if err := validator.Validate(openrtb_ext.BidderTpmn, json.RawMessage(invalidParam)); err == nil {
+			t.Errorf("Schema allowed unexpected params: %s", invalidParam)
+		}
+	}
+}
+
+var validParams = []string{
+	`{"inventoryId": 10000001}`,
+}
```

Comment:
nitpick - Generally, we read code from top to bottom. Declaring variable after it has already been used is not generally considered a good coding practice. Requesting you to please move this piece of code to the top(after imports statement) of params_test.go file .

Reply:
You are right. 
Since I'm not familiar with Go language, I referenced a lot of code from other companies and structured my code similarly. 
Many partners placed variables at the bottom of the code. 
That part also seems to need checking.

Reply:
update commit :
[1f22b27](https://github.com/prebid/prebid-server/pull/2947/commits/1f22b27a0bc801cd54d52dc7a9f1b9ffe8223894)